wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -92,13 +94,23 @@ gzip set app env sendResponse = app env $ \res ->
         ResponseFile{} | gzipFiles set == GzipIgnore -> sendResponse res
         _ -> if "gzip" `elem` enc && not isMSIE6 && not (isEncoded res) && (bigEnough res)
                 then
-                    case (res, gzipFiles set) of
-                        (ResponseFile s hs file Nothing, GzipCacheFolder cache) ->
-                            case lookup hContentType hs of
-                                Just m
-                                    | gzipCheckMime set m -> compressFile s hs file cache sendResponse
-                                _ -> sendResponse res
-                        _ -> compressE set res sendResponse
+                    let runAction x = case x of
+                            (ResponseFile s hs file Nothing, GzipPreCompressed nextAction) ->
```

Comment:
That's not the problem I was getting at. Presumably, in the case of `GzipPreCompressed GzipIgnore`, for non-`ResponseFile` responses the behavior should be identical to today's `GzipIgnore`. However, since the pattern match on `GzipIgnore` will fail, it will fall through to the default case and use `compressE`, which is not what is expected.